src/AutoRest.CSharp/Mgmt/Models/MgmtRestClientBuilder.cs

```diff
@@ -44,11 +40,28 @@ public MgmtRestClientBuilder(OperationGroup operationGroup)
         {
         }
 
-        public static IEnumerable<RequestParameter> GetMgmtParametersFromOperations(ICollection<Operation> operations) =>
-            operations
+        private static IReadOnlyList<RequestParameter> GetMgmtParametersFromOperations(ICollection<Operation> operations)
+        {
+            var parameters = operations
                 .SelectMany(op => op.Parameters.Concat(op.Requests.SelectMany(r => r.Parameters)))
                 .Where(p => p.Implementation == ImplementationLocation.Client)
-                .Distinct(new ParameterCompareer());
+                .Distinct(new ParameterCompareer()).ToList();
+            ValidateMgmtOperationParameters(parameters);
+            return parameters;
+        }
+
+        private static void ValidateMgmtOperationParameters(IReadOnlyList<RequestParameter> parameters)
+        {
+            var set = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "modelerfour:synthesized/host", "modelerfour:synthesized/api-version" };
+            foreach (var parameter in parameters)
+            {
+                var parameterName = parameter.Origin ?? string.Empty;
+                if (!set.Contains(parameterName))
+                {
+                    throw new InvalidOperationException($"{parameterName} should be method parameter for operation");
+                }
```

Comment:
we are using the origin of the parameter as its name in the error message here.
Also, would it help if we could include the information of the corresponding operation of this parameter?

Reply:
updated